Numeric.h

```diff
@@ -0,0 +1,184 @@
+#pragma once
+
+#include <functional>
+#include <cmath>
+#include <memory>
+
+template <typename T>
+struct Numeric
+{
+    using Type = T;
+
+    Numeric (Type t) : value (std::make_unique<Type>(t)) {}
+
+    Numeric(const Numeric& other) : value (std::make_unique<Type>(*other.value)) {}
+
+    ~Numeric() {}
+
+    operator double() const { return static_cast<double> (*value); }
+    operator float() const { return static_cast<float> (*value); }
+    operator int() const { return static_cast<int> (*value); }
+
+    Numeric& operator+= (Type rhs)
+    {
+        *value += rhs;
+        return *this;
+    }
+
+    Numeric& operator-= (Type rhs)
+    {
+        *value -= rhs;
+        return *this;
+    }
+
+    Numeric& operator*= (Type rhs)
+    {
+        *value *= rhs;
+        return *this;
+    }
+
+    template <typename RHS>
+    // PLEASE NOTE: Not sure if the intention was to need a template here, but I couldn't figure out how to do it without it.
+    Numeric& operator/= (RHS rhs)
+    {
+        if constexpr (std::is_same<int, Type>::value)
+        {
+            if constexpr (std::is_same<int, RHS>::value)
+            {
+                if (rhs == 0)
+                {
+                    std::cout << "error: integer division by zero is an error and will crash the program!" << std::endl;
+                    return *this;
+                }
+            }
+            else if (static_cast<float> (rhs) < std::numeric_limits<float>::epsilon())
+            {
+                std::cout << "can't divide integers by zero!" << std::endl;
+                return *this;
+            }
+        }
+        else if (static_cast<float> (rhs) < std::numeric_limits<float>::epsilon())
```

Comment:
don't cast to float here, use `RHS` as the template parameer to `numeric_limits`
- [ ] revise